docs/internals/sync.md

```diff
@@ -0,0 +1,215 @@
+---
+sidebar_position: 3
+---
+
+# State Sync
+
+State sync synchronizes the state of a lagging/divergent replica with the healthy cluster.
+
+State sync is used when when a lagging replica's log no longer intersects with the cluster's current log —
+[WAL repair](./vsr.md#protocol-repair-wal) cannot catch the replica up.
+
+(VRR refers to state sync as "state transfer", but we already have [transfers](../reference/transfers.md) elsewhere.)
+
+In the context of state sync, "state" refers to:
+1. the grid (LSM data; acquired blocks only)
+2. the superblock manifest
+3. the superblock free set
+4. the superblock client sessions (TODO: What about client replies?)
+5. the superblock `vsr_state.commit_min`
+6. the superblock `vsr_state.commit_min_checksum`
+
+The target of state sync is the latest checkpoint of the healthy cluster.
+When we catch up to the latest checkpoint (or very close to it), then we can transition back to a healthy state.
+
+## Glossary
+
+Replica roles:
+
+- _syncing replica_: A replica performing state sync. (Any step within *1*-*9* of the [sync algorithm](#algorithm))
+- _healthy replica_: A replica _not_ performing state sync — part of the active cluster.
+- _divergent replica_: A replica with a checkpoint that is (and can never be) canonical.
+
+Checkpoints:
+
+- [_checkpoint id_/_checkpoint identifier_](#checkpoint-identifier): Uniquely identifies a particular checkpoint reproducibly across replicas.
+- [_canonical checkpoint_](#canonical-checkpoint): Any checkpoint which either: _A_: has any ops committed atop it by the primary, or _B_: a view-change quorum has reached.
```

Comment:
This uses the term "checkpoint" inside the definition but checkpoint hasn't been defined or I'm missing it?

Reply:
I think we cover it in `vsr.md`. It isn't a new concept in the context of state sync, so it isn't defined here.

Reply:
Yup you're right it's there.